### PR TITLE
Add `cli lint` option to treat warnings as errors

### PR DESCRIPTION
**Pull request by 6543**
Add option treat warnings as errors for pipeline linter:

https://ci.woodpecker-ci.org/repos/3780/pipeline/22592/34 **FAILED SUCCESSFULLY**

### CODE REVIEW

**Comment by 6543** [comment]
should fail and pass after  #4372 merged ...

**Comment by 6543** [comment]
https://ci.woodpecker-ci.org/repos/3780/pipeline/22592/34 **FAILED SUCCESSFULLY**

File: .woodpecker/test.yaml [hunk 1]
```diff
@@ -40,6 +40,7 @@ steps:
       - go run go.woodpecker-ci.org/woodpecker/v2/cmd/cli lint
     environment:
       WOODPECKER_DISABLE_UPDATE_CHECK: true
+      WOODPECKER_WARNING_AS_ERROR: true
```
Comment (xoxys) [inline .woodpecker/test.yaml:43]: As this is an explicit linter config, can we rename it to `WOODPECKER_LINT_STRICT´?

Comment (6543) [reply .woodpecker/test.yaml:43]: sure, i have no clear preferences

Comment (6543) [reply .woodpecker/test.yaml]: 5547e0cdb784514b673e9c90aa36a230a8166f77

File: cli/lint/utils.go [hunk 2]
```diff
@@ -10,7 +10,7 @@ import (
 	pipeline_errors "go.woodpecker-ci.org/woodpecker/v2/pipeline/errors"
 )
 
-func FormatLintError(file string, err error) (string, error) {
+func FormatLintError(file string, err error, warningAsError bool) (string, error) {
```
Comment (xoxys) [inline cli/lint/utils.go:13]:
```suggestion
func FormatLintError(file string, err error, strict bool) (string, error) {
```

File: cli/lint/utils.go [hunk 3]
```diff
@@ -24,7 +24,7 @@ func FormatLintError(file string, err error) (string, error) {
 	for _, err := range linterErrors {
 		line := "  "
 
-		if err.IsWarning {
+		if !warningAsError && err.IsWarning {
```
Comment (xoxys) [inline cli/lint/utils.go:27]:
```suggestion
		if !strict && err.IsWarning {
```

**Review by xoxys — APPROVED** [review]
some more nits

File: .woodpecker/test.yaml (hunk 1 above)
Comment (xoxys) [reply .woodpecker/test.yaml]: Thanks!

**Pull request merged**